src/applications/disability-benefits/all-claims/components/OfficialReport.jsx

```diff
@@ -0,0 +1,304 @@
+  const isAdd = !!searchParams.get('add');
+  const index = parseInt(props.pagePerItemIndex, 10);
+
+  const [tempData, setTempData] = useState(props.data || {});
+  const [showModal, setShowModal] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handlers = {
+    shouldShowPoliceDataModal,
+    onModalOrContinue: ({ formData }) => {
+      if (shouldShowPoliceDataModal(formData)) {
+        setShowModal(true);
+        return;
+      }
+
+      if (!formDomRef.current) return;
+      setTempData(formData);
+      formDomRef.current.submitForm();
+    },
+    onSubmit: ({ formData }) => {
+      if (handlers.shouldShowPoliceDataModal(formData)) {
+        setShowModal(true);
+        setTempData(formData);
+        return;
+      }
+
+      props.onChange(formData);
+      props.onSubmit({ formData });
+    },
+    onCancelModal: () => {
+      setShowModal(false);
+      if (!props.data) return;
+      const restoredData = {
+        ...tempData,
+        otherReports: { ...tempData.otherReports, police: true },
+      };
+
+      setTempData(restoredData);
+    },
+    onCancelAlert: () => {
+      setShowAlert(false);
+    },
+    removePoliceReport: () => {
+      setShowModal(false);
+
+      const updatedData = { ...tempData };
+      POLICE_REPORT_LOCATION_FIELDS.forEach(field => {
+        updatedData[field] = '';
+      });
+      updatedData.otherReports = { ...updatedData.otherReports, police: false };
+
+      setTempData(updatedData);
+      props.onChange(updatedData);
+      setShowAlert(true);
+    },
+    handleChange: newData => {
+      setTempData(newData);
+    },
+  };
+
+  const { schema, uiSchema } = useEditOrAddForm({
+    isEdit,
+    schema: isRelatedToMST(props.fullData)
+      ? officialReportMst.schema
+      : officialReport.schema,
+    uiSchema: isRelatedToMST(props.fullData)
+      ? officialReportMst.uiSchema
+      : officialReport.uiSchema,
+    data: tempData,
+    fullData: props.fullData,
+    onChange: handlers.handleChange,
+    onSubmit: handlers.onModalOrContinue,
+    index: props.pagePerItemIndex ? index : null,
+    arrayPath: props.arrayBuilder.arrayPath,
+  });
+
+  useEffect(
+    () => {
+      if (showModal && modalRef.current) {
+        setTimeout(() => {
+          const modalHeading = modalRef.current.shadowRoot?.querySelector(
+            '#heading',
+          );
+          if (modalHeading) {
+            modalHeading.focus({ preventScroll: true });
+          }
+        }, 0);
+      }
+    },
+    [showModal],
+  );
+  useEffect(
+    () => {
+      if (showAlert && alertRef.current) {
+        alertRef.current.focus();
+      }
+    },
+    [showAlert],
+  );
+
+  if (!props.onReviewPage && !isEdit && !isAdd) {
+    const path = props.arrayBuilder.summaryRoute;
+    props.goToPath(path);
+    return null;
+  }
+
+  if (props.onReviewPage || (isEdit && !schema)) {
+    return null;
+  }
+
+  const NavButtons = props.NavButtons || FormNavButtons;
```

Comment:
I'd doc why we need to allow us to pass in the buttons as props (presumably an array builder thing)

Reply:
Yep, adopted from `ArrayBuilderItemPage`, where there also is no explanation provided.